app/components/Showcases/Barter.jsx

```diff
@@ -455,45 +466,49 @@ export default class Barter extends Component {
                     to_account: this.state.escrow_account.get("id"),
                     amount: escrow_payment,
                     asset: "1.3.0",
-                    memo: this.state.memo
-                        ? new Buffer(this.state.memo, "utf-8")
-                        : this.state.memo,
+                    memo: escrowMemo ? new Buffer(escrowMemo, "utf-8") : null,
                     feeAsset: "1.3.0",
                     propose_account: proposer
                 });
             }
         }
 
-        this.state.from_barter.forEach(item => {
+        this.state.from_barter.forEach((item, index) => {
             const asset = item.from_asset;
             let amount = item.from_amount;
             sendAmount = new Asset({
                 real: amount,
                 asset_id: asset.get("id"),
                 precision: asset.get("precision")
             });
+
+            let fromBarterMemo =
+                this.state.memo["from_barter"][index] &&
```

Comment:
Is there a specific reason you put this in a separate memo state instead of including it in the from_barter state as a new memo key?

Reply:
Only to keep all memos in separate obj.